public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.test.tsx

```diff
@@ -165,4 +166,50 @@ describe('QueryEditor', () => {
       expect(await screen.findByText('*')).toBeInTheDocument();
     });
   });
+
+  describe('when dynamic labels feature toggle is enabled', () => {
+    it('shoud render label field', async () => {
+      await act(async () => {
+        const props = setup();
+        const originalValue = config.featureToggles.cloudWatchDynamicLabels;
+        config.featureToggles.cloudWatchDynamicLabels = true;
+
+        const result = render(
+          <MetricsQueryEditor
+            {...props}
+            query={{ ...props.query, refId: 'A', alias: 'Period: {{period}} InstanceId: {{InstanceId}}' }}
+          />
+        );
+
+        expect(screen.getByText('Label')).toBeInTheDocument();
+        expect(screen.queryByText('Alias')).toBeNull();
+        expect(result.container.querySelector('#A-cloudwatch-metric-query-editor-label')).toHaveValue(
```

Comment:
Was not able to get this to work:
```typescript 
expect(screen.getByLabelText('Label')).toHaveValue(
          "Period: ${PROP('Period')} InstanceId: ${PROP('Dim.InstanceId')}"
        );
```

Not sure why? This is an example of the rendered HTML

```html
<div class="css-1hjz9ft">
   <label class="css-11h8yic" for="A-cloudwatch-metric-query-editor-label">
      Label<span class="css-18lesqj"> - optional</span>
      <div class="css-1vzus6i-Icon">
         <svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="14" height="14" class="css-3vp1c4">
            <path d="M12,2A10,10,0,1,0,22,12,10.01114,10.01114,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8.00917,8.00917,0,0,1,12,20Zm0-8.5a1,1,0,0,0-1,1v3a1,1,0,0,0,2,0v-3A1,1,0,0,0,12,11.5Zm0-4a1.25,1.25,0,1,0,1.25,1.25A1.25,1.25,0,0,0,12,7.5Z"></path>
         </svg>
      </div>
   </label>
   <span class="css-djpfq6"></span>
   <div>
      <div class="css-xcstkt-input-wrapper" data-testid="input-wrapper">
         <div class="css-1w5c5dq-input-inputWrapper"><input class="css-yn255a-input-input" id="A-cloudwatch-metric-query-editor-label" placeholder="auto" value="Period: ${PROP('Period')} InstanceId: ${PROP('Dim.InstanceId')}"></div>
      </div>
   </div>
</div>
```